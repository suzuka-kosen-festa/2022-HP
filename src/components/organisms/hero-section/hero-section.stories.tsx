import type { ComponentStoryObj, ComponentMeta } from "@storybook/react";
import { HeroSection } from ".";

type T = typeof HeroSection;
type Story = ComponentStoryObj<T>;

const data = {
  date: "2022.10.30-31",
  title: "第57回鈴鹿高専祭",
};

export default { args: { date: data.date, title: data.title }, component: HeroSection } as ComponentMeta<T>;

export const Default: Story = {};
