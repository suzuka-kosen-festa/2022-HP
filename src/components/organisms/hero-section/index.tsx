import type { FC } from "react";
import tw, { css } from "twin.macro";
import type { HeroSectionProperties } from "../../../models";
import { Text } from "../../atoms/text";

const HeroSectionContainer = tw.header`flex flex-col`;

const Background = tw.div`flex flex-col items-center justify-center space-y-[calc(200vw / 63)] w-full h-[calc(200vw / 3)] lg:h-screen bg-center bg-cover`;

const Logo = tw.img`w-[calc(8200vw / 189)]`;

const HeroSection: FC<HeroSectionProperties> = ({ date, title, ...rest }) => (
  <HeroSectionContainer {...rest}>
    <Background
      css={css`
        background-image: linear-gradient(rgba(34, 31, 31, 0.9), rgba(34, 31, 31, 0.9)), url("/statics/gogo.webp");
      `}
    >
      <Text css={tw`text-[calc(16px + 16 * ((100vw - 378px) / 1134))]`}>{title}</Text>
      <Logo src="/statics/logo.webp" alt="第57回鈴鹿高専祭、Re:ROADのロゴ" width={1212} height={247} />
      <Text css={tw`text-[calc(16px + 16 * ((100vw - 378px) / 1134))]`}>{date}</Text>
    </Background>
  </HeroSectionContainer>
);

export { HeroSection };
