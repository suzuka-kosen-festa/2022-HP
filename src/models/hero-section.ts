import type { ComponentProps, ReactHTML } from "react";

export type HeroSectionProperties = Omit<ComponentProps<ReactHTML["header"]>, "children"> & {
  date: string;
  title: string;
};
