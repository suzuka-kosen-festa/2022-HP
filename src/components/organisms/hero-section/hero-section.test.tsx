import { composeStories } from "@storybook/testing-react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import * as stories from "./hero-section.stories";

const { Default } = composeStories(stories);

const options = {
  name: "第57回鈴鹿高専祭",
};

describe("(components) organisms/hero-section", () => {
  test("to be organisms", () => {
    const { container } = render(<Default />);
    expect(container).toBeOrganism();
  });
  test("to be [role=region]", () => {
    const { getByRole } = render(<Default />);
    expect(getByRole("region", options));
  });
  test("take snap shot", () => {
    const { container } = render(<Default />);
    expect(container).toMatchSnapshot();
  });
});
