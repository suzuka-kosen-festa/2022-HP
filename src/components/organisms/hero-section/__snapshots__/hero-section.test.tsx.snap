// Vitest Snapshot v1

exports[`(components) organisms/hero-section > take snap shot 1`] = `
<div>
  <header
    class="css-j7qwjs"
  >
    <div
      class="css-bg20hz-HeroSection"
    >
      <p
        class="css-8shmws-HeroSection"
      >
        第57回鈴鹿高専祭
      </p>
      <img
        alt="第57回鈴鹿高専祭、Re:ROADのロゴ"
        class="css-1ysdsd"
        height="247"
        src="/statics/logo.webp"
        width="1212"
      />
      <p
        class="css-8shmws-HeroSection"
      >
        2022.10.30-31
      </p>
    </div>
  </header>
</div>
`;
